docs/download.md

```diff
@@ -85,25 +86,24 @@ CUDA 11.8 through CUDA 12.0.
 * Download the [PUB_KEY](https://keys.openpgp.org/search?q=sw-spark@nvidia.com).
 * Import the public key: `gpg --import PUB_KEY`
 * Verify the signature for Scala 2.12 jar:
-    `gpg --verify rapids-4-spark_2.12-25.02.1.jar.asc rapids-4-spark_2.12-25.02.1.jar`
+    `gpg --verify rapids-4-spark_2.12-25.04.0.jar.asc rapids-4-spark_2.12-25.04.0.jar`
 * Verify the signature for Scala 2.13 jar:
-    `gpg --verify rapids-4-spark_2.13-25.02.1.jar.asc rapids-4-spark_2.13-25.02.1.jar`
+    `gpg --verify rapids-4-spark_2.13-25.04.0.jar.asc rapids-4-spark_2.13-25.04.0.jar`
 
 The output of signature verify:
 
 	gpg: Good signature from "NVIDIA Spark (For the signature of spark-rapids release jars) <sw-spark@nvidia.com>"
 
 ### Release Notes
-* Support the Spark functions Bin and TruncDate
-* Support group-limit optimization for ROW_NUMBER
-* Improve Spark metrics: Print the batch size information to executor log
-* Refine filter push down to avoid double evaluation
-* Grab the GPU Semaphore when reading cached batch data with the GPU to avoid a GPU OOM case
-* Add an option to disable measuring buffer copy to improve large shuffle large partition serialization
-* For updates on RAPIDS Accelerator Tools, please visit [this link](https://github.com/NVIDIA/spark-rapids-tools/releases)
-* Upgraded statically linked CUDA toolkit to 12.8, which includes support for GB100 GPUs
-
-Note: There is a known issue in the 25.02.1 release when decompressing gzip files on H100 GPUs.
+* Support group by on binary type for databricks
+* Support ArrayPosition function for databricks
+* Support Databricks 14.3 ML LTS
+* Refine split-retry logs when out of memory happens to expose the real reason
+* Allow BigSizedJoinIterator#buildPartitioner to produce more sub-partitions to avoid CudfColumnSizeOverflowException
+* Fix a hanging issue for Python UDF in spark 4.0.0
+* Fix a device memory leak when re-partitioning in GpuSubPartitionHashJoin
+
+Note: There is a known issue in the 25.04.0 release when decompressing gzip files on H100 GPUs.
```

Comment:
```suggestion
* Support approx_count_distinct
* Support group by on binary type 
* Support ArrayPosition function 
* Support Slice
* Support Databricks 14.3 ML LTS
* Enable Hive text writer
* Refine split-retry logs when out of memory happens to expose the real reason
* Allow BigSizedJoinIterator#buildPartitioner to produce more sub-partitions to avoid CudfColumnSizeOverflowException

Note: There is a known issue in the 25.04.0 release when decompressing gzip files on H100 GPUs.
```